### PR TITLE
storage: Polish interior layout of the side panels

### PR DESCRIPTION
**Pull request by mvollmer**
The name now uses the whole width of the row.

- Use a flexbox instead of a table (but keep the table for easy
  borders and highlighting).

- Remove the icon since every row in a panel has the same icon anyway.

- Add the filename of the block device.

### For release notes

The side panel of the Storage page does not truncate names anymore and shows names in `/dev`.

image

### CODE REVIEW

**Comment by mvollmer** [comment]
Old on the left, new on the right:

image image

**Comment by martinpitt** [comment]
The [selenium storage](https://209.132.184.41:8493/logs/pull-11825-20190514-124848-619d2fb5-selenium-chrome/log.html#10) test is broken on all browsers. 

There's also several instances of [iscsi failures](https://209.132.184.41:8493/logs/pull-11825-20190514-124848-619d2fb5-verify-rhel-8-0-distropkg/log.html#170), but it's not clear that this is a regression - I'll retry to compare.

**Comment by mvollmer** [comment]
> The selenium storage test is broken on all browsers.

Should be fixed now by using a more specific xpath.  With this PR, we had multiple `div`s that contain the expected string.

**Comment by mvollmer** [comment]
> There's also several instances of iscsi failures, but it's not clear that this is a regression - I'll retry to compare.

That was a regression, the delete button was missing completely.

**Comment by martinpitt** [comment]
@mvollmer: This needs to be rebased. @andreasn: Are you ok with the design here?

**Comment by mvollmer** [comment]
Rebased, to get fresh test runs.

**Comment by andreasn** [comment]
I think I'm OK with these changes, but I want to get feedback from @garrett as well.

**Comment by garrett** [comment]
Apologies; I thought I wrote this comment before. Perhaps I just said it on IRC, in a conversation with @andreasn, or forgot to hit the big green button to post it:

I am not a fan of this.

- It makes the elements look like big, noisy blobs of text
- I'm not sure how useful the read/write stats are
- Saying "Hard Disk" is redundant

Suggestions:
- Use typographical hierarchy, using text sizes and/or weights (the icon helped with this before, acting as an anchor for each item)
- Re-think disk IO stats (perhaps for another PR), with a better moving average (smoothing) or sparklines, with stats only appearing when relevant
- Drop the "Hard Disk" label; it's implied — optical is the exceptional disk (does it state the size when it is mounted?)

This isn't ready yet. :-1:

**Comment by mvollmer** [comment]
@garrett, thanks!  It looks like this now:

image

But let's hold this until your PF4 work is merged.

**Comment by mvollmer** [comment]
> I'm not sure how useful the read/write stats are

Me neither.  They are mostly there "because we can".  Removing them allows us to remove some extra polling that we do to get them, and the stats are in fact a major reason why the page doesn't scale to hundreds of drives.  So I have removed them.

**Comment by andreasn** [comment]
> because we can

Yup, I can confirm this. Neat idea, but haven't found it useful. Let's kill them.

**Comment by andreasn** [comment]
We could also just move the read and write to the drive's detail page.

**Comment by garrett** [comment]
@mvollmer: This is _much_ better! Nice!

I think the bolding might be a bit much in context. Once the PF4ish PR lands, we can tweak this one a bit by making the headings normal and just lighten the details to the lower priority informational text color.

Also, how important is the complete name? Do we want it to wrap if there's no space, or is it fine enough being truncated (as-is)?

**Comment by mvollmer** [comment]
> I think the bolding might be a bit much in context. Once the PF4ish PR lands, we can tweak this one a bit by making the headings normal and just lighten the details to the lower priority informational text color.

Yeah, I have a version for PF4 already:

image

I think we need to make the sidebar wider, actually.
 
> Also, how important is the complete name? Do we want it to wrap if there's no space, or is it fine enough being truncated (as-is)?

I can't say.  For me personally, the device name /dev/blah helps a lot to select the right one.

**Comment by martinpitt** [comment]
> For me personally, the device name /dev/blah helps a lot to select the right one.

Absolutely, it's still the key identifier under Linux. *if* it has to truncate, then the "/dev/" bit at the front. It's absolutely useless to truncate it at the end.

**Comment by garrett** [comment]
> I think we need to make the sidebar wider, actually.

I think we need to redo the layout for the page. The order doesn't make sense on mobile (or on screen readers).

I'd suggest using CSS grid to lay it out more properly. I'm happy to do this. However, while we're rethinking the layout, we might also want to adjust the design a bit.

**Comment by garrett** [comment]
Based on the screenshot, I'm happy with it as-is for this PR and consider all the other talked about changes candidates for follow-up PRs.

(I'm checking out the PR now to look at it on my own system.)

File: pkg/storaged/storage.css [hunk 1]
```diff
@@ -724,3 +664,35 @@ td.storage-action {
...
+#storage .storage-sidebar .table {
+    table-layout: fixed;
+}
+
+.sidepanel-row {
+    display: flex;
+    flex-direction: row;
+}
+
+.sidepanel-row :nth-child(1) {
```
Comment (garrett) [inline pkg/storaged/storage.css]: Don't use `:nth-child()`; add a class in the HTML (React) and use the class in CSS instead.

File: pkg/storaged/storage.css [hunk 2]
```diff
@@ -724,3 +664,35 @@ td.storage-action {
...
+    flex: 1;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+}
+
+.sidepanel-row :nth-child(2) {
```
Comment (garrett) [inline pkg/storaged/storage.css]: Same here, don't use `:nth-child()`.

File: pkg/storaged/storage.css [hunk 3]
```diff
@@ -724,3 +664,35 @@ td.storage-action {
...
+    align-self: center;
+    white-space: nowrap;
+}
+
+.sidepanel-row .spinner {
+    display: inline-block;
+}
+
+.sidepanel-row .rate {
+    width: 15c;
```
Comment (garrett) [inline pkg/storaged/storage.css]: This isn't a valid unit. Perhaps it's supposed to be `15ch`?

(This one is probably my fault, actually. I think I worked on this CSS. Well, oops. The intention is to not have the width change when the data # changes.)

File: pkg/storaged/storage.css (hunk 1 above)
Comment (mvollmer) [reply pkg/storaged/storage.css]: Done.

File: pkg/storaged/storage.css (hunk 2 above)
Comment (mvollmer) [reply pkg/storaged/storage.css]: Done.

File: pkg/storaged/storage.css (hunk 3 above)
Comment (mvollmer) [reply pkg/storaged/storage.css]: We don't need this anymore now that the rate is no longer displayed.

**Comment by mvollmer** [comment]
> Also, things are in empty <div>s

One empty div is left, but that's to get a multi-line layout so I hope it is ok.

**Comment by mvollmer** [comment]
@garrett, please have another look.

**Comment by garrett** [comment]
At some widths, the disks overlap:
Screenshot_2019-07-24 Storage - Rain

Close-up crop:

Screenshot_2019-07-24 Storage - Rain-overlap


When I expand my browser to be wide, I see this:

Screenshot_2019-07-24 Storage - Rain(1)

**Comment by garrett** [comment]
I guess I should do some layout work on these so they present the information better and wrap if needed. It's probably just a matter of modifying the local CSS.

**Comment by mvollmer** [comment]
> I guess I should do some layout work on these so they present the information better and wrap if needed. 

@garrett, I will try my hand at flexbox, so that we get some wrapping like this:

```
XXX GiB Block Device
                /dev/fedora-rain/foo
```

**Comment by mvollmer** [comment]
> @garrett, I will try my hand at flexbox,

Done.

**Comment by garrett** [comment]
Considering the layout of part of the storage page really makes me want for us to redo the main layout. But that's for a future PR. :wink:

File: pkg/storaged/client.js [hunk 4]
```diff
@@ -171,10 +171,6 @@ client.swap_sizes = instance_sampler([ { name: "swapdev.length" },
     { name: "swapdev.free" },
 ], "direct");
 
-client.blockdev_io = instance_sampler([ { name: "block.device.read", derive: "rate" },
```
Comment (martinpitt) [inline pkg/storaged/client.js]: The commit message could explain the reason for dropping these.

**Pull request merged**